cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -0,0 +1,455 @@
+ * @param end The end of the validity indicator sequence
+ * @return rmm::device_buffer Contains a bitmask where bits are set for every
+ * element in `[begin,end)` that evaluated to `true`.
+ *---------------------------------------------------------------------------**/
+template <typename ValidityIterator>
+rmm::device_buffer make_null_mask(ValidityIterator begin,
+                                  ValidityIterator end) {
+  auto null_mask = make_null_mask_vector(begin, end);
+  return rmm::device_buffer{
+      null_mask.data(), null_mask.size() * sizeof(decltype(null_mask.front()))};
+}
+
+/**---------------------------------------------------------------------------*
+ * @brief Given the range `[begin,end)`, converts each value to a string and
+ * then creates a packed vector of characters for each string and a vector of
+ * offsets indicating the starting position of each string.
+ *
+ * @tparam StringsIterator A `std::string` must be constructible from
+ * dereferencing a `StringsIterator`.
+ * @tparam ValidityIterator Dereferencing a ValidityIterator must be
+ * convertible to `bool`
+ * @param begin The beginning of the sequence of values to convert to strings
+ * @param end The end of the sequence of values to convert to strings
+ * @param v The beginning of the validity indicator sequence
+ * @return std::pair containing the vector of chars and offsets
+ *---------------------------------------------------------------------------**/
+template <typename StringsIterator, typename ValidityIterator>
+auto make_chars_and_offsets(StringsIterator begin, StringsIterator end,
+                            ValidityIterator v) {
+  std::vector<char> chars{};
+  std::vector<int32_t> offsets(1, 0);
+  for (; begin < end; ++begin) {
+    std::string tmp = (*v++) ? std::string(*begin) : std::string{};
+    chars.insert(chars.end(), std::cbegin(tmp), std::cend(tmp));
+    offsets.push_back(offsets.back() + tmp.length());
+  }
+  return std::make_pair(chars, offsets);
+};
+}  // namespace detail
+
+/**---------------------------------------------------------------------------*
+ * @brief `column_wrapper` derived class for wrapping columns of fixed-width
+ * elements.
+ *
+ * @tparam Element The fixed-width element type
+ *---------------------------------------------------------------------------**/
+template <typename Element>
+class fixed_width_column_wrapper : public detail::column_wrapper {
+  static_assert(cudf::is_fixed_width<Element>(),
+                "Unexpected non-fixed width type.");
+
+ public:
+  /**---------------------------------------------------------------------------*
+   * @brief Construct a non-nullable column of the fixed-width elements in the
+   * range `[begin,end)`.
+   *
+   * Example:
+   * ```c++
+   * // Creates a non-nullable column of INT32 elements with 5 elements: {0, 1, 2, 3, 4}
+   * auto elements = make_counting_transform_iterator(0, [](auto i){return i;});
```

Comment:
The reader is left wondering why use a transform iterator for this example? Maybe make it slightly more complicated -- make it 5 even values:
```suggestion
   * auto elements = make_counting_transform_iterator(0, [](auto i){return i*2;});
```